libs/features/google-login.js

```diff
@@ -3,10 +3,11 @@ const GOOGLE_ID = '530526366930-l874a90ipfkn26naa71r010u8epp39jt.apps.googleuser
 const PLACEHOLDER = 'feds-googleLogin';
 const WRAPPER = 'feds-profile';
 
-const onToken = async (getMetadata, data) => {
+const onToken = async (getMetadata, data, getConfig) => {
   let destination;
+  const config = getConfig();
   try {
-    destination = new URL(getMetadata('google-login-redirect'))?.href;
+    destination = typeof config.googleLoginURLCallback === 'function' ? new URL(config.googleLoginURLCallback()) : new URL(getMetadata('google-login-redirect'))?.href;
```

Comment:
why do we need a new method that returns an url? can't the existing metadata be used and updated?

Reply:
@narcis-radu that hook already existed before this PR. I just simplified it a little while originally adding two other hooks. 
As for the why we need a callback, sure. 
currently in Express when we display the google yolo popup, we redirect the user to the link that is in the main CTA or floating button in Express. This means that the redirect link is dynamic and is only "known", once the page has loaded. I'd like to pass a method, so that the method can resolve and figure out what the link should be. Setting metadata doesn't quite work, because I don't know when the page loads what the link will be. 
Does that make sense or do you need a few more details?

Reply:
It makes sense, the only concern is that we're adding too much complexity - the URL can be specified as a metadata or as a callback. I was wondering maybe you can do the following: in your project, update the metadata as soon as you have the URL ready, this way you wouldn't need to set a callback in Milo. Would that work ?

Reply:
I completley agree with you, but not really unfortunately. I only know what the URL is for certain once all the sections have been processed. So I'd either need an event or a hook which tells me that runs once all the sections have finished processing and sets the metadata then, or I need a callback like this which resolves at the precise point in time that I need it.

Reply:
actually hang on, I might have spoken too soon. Let me check 👍

Reply:
ok, no , it doesn't work. If it was just Express blocks I needed to handle your solution would work. I really like it actually. 
The problem is that for milo blocks that load in I can't set this metadata in between while they're loading (unless I had a hook or something), so this method has to resolve as said once the page has finished loading.